src/category_theory/limits/has_limits.lean

```diff
@@ -463,10 +476,25 @@ end lim_functor
 /--
 We can transport limits of shape `J` along an equivalence `J ≌ J'`.
 -/
-lemma has_limits_of_shape_of_equivalence {J' : Type v} [small_category J']
+lemma has_limits_of_shape_of_equivalence {J' : Type u₂} [category.{v₂} J']
   (e : J ≌ J') [has_limits_of_shape J C] : has_limits_of_shape J' C :=
 by { constructor, intro F, apply has_limit_of_equivalence_comp e, apply_instance }
 
+variable (C)
+
+/--
+`has_limits_of_size.{v u} C` tries to obtain `has_limits_of_size.{v u} C`
+from some other `has_limits_of_size C`.
+-/
+lemma has_limits_of_size_shrink [has_limits_of_size.{(max v₁ v₂) (max u₁ u₂)} C] :
+  has_limits_of_size.{v₁ u₁} C :=
+⟨λ J hJ, by exactI has_limits_of_shape_of_equivalence
+  (ulift_hom_ulift_category.equiv.{v₂ u₂} J).symm⟩
+
+@[priority 100]
+instance has_smallest_limits_of_has_limits [has_limits C] :
+  has_limits_of_size.{0 0} C := has_limits_of_size_shrink.{0 0} C
+
```

Comment:
We know that if `C` has limits of some size, then it has limits of smaller sizes. But this could not be made an instance. The best instance we could make is `has_limits C \to has_limits_of_size.{0 0} C`.